hugegraph-core/src/main/java/org/apache/hugegraph/backend/tx/GraphTransaction.java

```diff
@@ -1696,12 +1719,9 @@ private <T extends HugeElement> Iterator<T> filterUnmatchedRecords(
                 return false;
             }
             // Process results that query from left index or primary-key
-            if (query.resultType().isVertex() == elem.type().isVertex() &&
-                !rightResultFromIndexQuery(query, elem)) {
-                // Only index query will come here
-                return false;
-            }
-            return true;
+            // Only index query will come here
```

Comment:
the logic is not equivalent here, I think maybe the origin code is more readable

Reply:
fixed

Reply:
`// Only index query will come here` only when returning false, prefer to revert it